Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -831,6 +833,25 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         // Bin dir will be set when developing swiftpm without building all of the runtimes.
         return resources.binDir ?? resources.libDir.appending(version.runtimeSubpath)
     }
+
+    /// Returns path to the manifest database inside the given cache directory.
+    private static func manifestCacheDBPath(_ cacheDir: AbsolutePath) -> AbsolutePath {
+        return cacheDir.appending(component: "manifest.db")
+    }
+
+    func createCacheIfNeeded() throws {
+        // Return if we have already created the cache.
+        guard self.cache == nil else { return }
+        guard let manifestCacheDBPath = cacheDir.flatMap({ Self.manifestCacheDBPath($0) }) else { return }
+        self.cache = try SQLiteBackedPersistentCache(cacheFilePath: manifestCacheDBPath)
+    }
+
+    public func resetCache() throws {
+        guard let manifestCacheDBPath = cacheDir.flatMap({ Self.manifestCacheDBPath($0) }) else { return }
+        self.cache = nil
```

Comment:
It's slightly worrisome that an explicit action such as closing a SQLite database is triggered by the `deinit()` of the cache.  If it's held alive for whatever reason, it means the database won't be closed.  Like file handles and things like that, this might be a case in which a specific `close()` method is warranted (which should be idempotent and could be called by `deinit()` as well, in case a client forgets to do it).

Reply:
Exposing `close()` on `SQLiteBackedPersistentCache` makes sense but I think this is fine for now since the object is not exposed out of manifest loader.

Reply:
Makes sense.  It was mostly a more forward-looking question.